### PR TITLE
[20820] Make security private

### PR DESCRIPTION
**Pull request by elianalf**
## Description
This pr makes security private, in particular:
* logging
* exception
* cryptography
* common
* authentication
* access control

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- N/A Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- N/A Any new/modified methods have been properly documented using Doxygen. 
- N/A Any new configuration API has an equivalent XML API (with the corresponding XSD extension) 
- ❌ Changes are backport compatible: they do **NOT** break ABI nor change library core behavior. 
- ❌ Changes are API compatible. 
- N/A New feature has been added to the `versions.md` file (if applicable).
- N/A New feature has been documented/Current behavior is correctly described in the documentation. 
    
- N/A Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: src/cpp/rtps/security/SecurityPluginFactory.h [hunk 1]
```diff
@@ -18,13 +18,13 @@
 #ifndef _RTPS_SECURITY_SECURITYPLUGINFACTORY_H_
 #define _RTPS_SECURITY_SECURITYPLUGINFACTORY_H_
 
-#include <fastdds/rtps/security/authentication/Authentication.h>
-#include <fastdds/rtps/security/accesscontrol/AccessControl.h>
-#include <fastdds/rtps/security/cryptography/Cryptography.h>
-#include <fastdds/rtps/security/logging/Logging.h>
-#include <fastdds/rtps/attributes/PropertyPolicy.h>
```
Comment (JesusPoderoso) [inline src/cpp/rtps/security/SecurityPluginFactory.h:21]:
```suggestion
```

File: src/cpp/security/accesscontrol/Permissions.cpp [hunk 2]
```diff
@@ -32,12 +32,12 @@
 #include <openssl/err.h>
 #include <openssl/obj_mac.h>
 
+
```
Comment (JesusPoderoso) [inline src/cpp/security/accesscontrol/Permissions.cpp:35]:
```suggestion
```

File: test/mock/rtps/SecurityPluginFactory/rtps/security/SecurityPluginFactory.h [hunk 3]
```diff
@@ -18,13 +18,13 @@
 #ifndef _RTPS_SECURITY_SECURITYPLUGINFACTORY_H_
 #define _RTPS_SECURITY_SECURITYPLUGINFACTORY_H_
 
-#include <fastdds/rtps/attributes/PropertyPolicy.h>
-#include <fastdds/rtps/security/accesscontrol/AccessControl.h>
-#include <fastdds/rtps/security/authentication/Authentication.h>
-#include <fastdds/rtps/security/cryptography/Cryptography.h>
-#include <fastdds/rtps/security/logging/Logging.h>
```
Comment (JesusPoderoso) [inline test/mock/rtps/SecurityPluginFactory/rtps/security/SecurityPluginFactory.h:21]:
```suggestion
```

File: test/unittest/rtps/security/SecurityTests.hpp [hunk 4]
```diff
@@ -17,18 +17,18 @@
 
 #include <gtest/gtest.h>
 
+
```
Comment (JesusPoderoso) [inline test/unittest/rtps/security/SecurityTests.hpp:20]:
```suggestion
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
Most of the changes may dissapear also while passing the linter (uncrustify action failed)

**Review by JesusPoderoso — DISMISSED** [review]
LGTM with green CI

**Review by JesusPoderoso — DISMISSED** [review]
LGTM with green CI

**Comment by elianalf** [comment]
@richiprosima please test_3 this

**Review by JesusPoderoso — APPROVED** [review]
LGTM with green CI

**Comment by elianalf** [comment]
@richiprosima please test_3 this

**Pull request merged**